### PR TITLE
Try to fix Windows build

### PR DESCRIPTION
**Pull request by dmitrio95**
An attempt to fix AppVeyor build.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -242,14 +242,15 @@ else (APPLE)
       endif(BUILD_64 STREQUAL "ON")
    # Set compiler options for VS2017 toolchain.
    # Note: /D_CRT_SECURE_NO WARNINGS disables warnings when using "non-secure" library functions like sscanf...
-      set(CMAKE_CXX_FLAGS                "/MP /DWIN32 /D_WINDOWS /GR /EHsc /D_UNICODE /DUNICODE /D_CRT_SECURE_NO_WARNINGS /execution-charset:utf-8 /source-charset:utf-8")
+      set(CMAKE_CXX_FLAGS                "/MP /MT /DWIN32 /D_WINDOWS /GR /EHsc /D_UNICODE /DUNICODE /D_CRT_SECURE_NO_WARNINGS /execution-charset:utf-8 /source-charset:utf-8")
+      set(CMAKE_C_FLAGS                  "/MP /MT /DWIN32 /D_WINDOWS")
       if (USE_SSE)
          # If USE_SSE is set, enable use of SSE instructions to compiler.
          set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} /arch:SSE")
```
Comment (Jojo-Schmitz) [inline CMakeLists.txt]: Wouldn't `set(CMAKE_C_FLAGS    "${CMAKE_C_FLAGS} /arch:SSE")` be needed here?

Comment (dmitrio95) [reply CMakeLists.txt]: Probably so, I'll add it there if this approach with C flags finally works.

Comment (Jojo-Schmitz) [reply CMakeLists.txt]: Yay! It does!
And locally on my PC too!

File: CMakeLists.txt [hunk 2]
```diff
@@ -242,15 +242,18 @@ else (APPLE)
...
       endif (USE_SSE)
-      set(CMAKE_CXX_FLAGS_DEBUG          "/permissive- /std:c++14 /W4 /Zi /Ob0 /Od /RTC1")
-      set(CMAKE_CXX_FLAGS_RELEASE        "/permissive- /std:c++14 /W4 /O2 /Ob2 /DNDEBUG /DQT_NO_DEBUG /DQT_NO_DEBUG_OUTPUT")
-      set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/permissive- /std:c++14 /W4 /Zi /O2 /Ob1 /DNDEBUG /DQT_NO_DEBUG")
+      set(CMAKE_CXX_FLAGS_DEBUG          "/MT /permissive- /std:c++14 /W4 /Zi /Ob0 /Od /RTC1")
+      set(CMAKE_CXX_FLAGS_RELEASE        "/MT /permissive- /std:c++14 /W4 /O2 /Ob2 /DNDEBUG /DQT_NO_DEBUG /DQT_NO_DEBUG_OUTPUT")
+      set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MT /permissive- /std:c++14 /W4 /Zi /O2 /Ob1 /DNDEBUG /DQT_NO_DEBUG")
+      set(CMAKE_C_FLAGS_DEBUG            "/MT /Zi /Ob0 /Od /RTC1")
+      set(CMAKE_C_FLAGS_RELEASE          "/MT /O2 /Ob2 /DNDEBUG")
+      set(CMAKE_C_FLAGS_RELWITHDEBINFO   "/MT /Zi /O2 /Ob1 /DNDEBUG")
```
Comment (Jojo-Schmitz) [inline CMakeLists.txt]: I guess we'd better also add the `/DQT_NO_DEBUG` and `/DQT_NO_DEBUG_OUTPUT` here

Comment (dmitrio95) [reply CMakeLists.txt]: Would they make sense for C source code? It doesn't look like Qt works in C language at all.

Comment (Jojo-Schmitz) [reply CMakeLists.txt]: Hmm, OK, maybe. Still, it is just defines, so they won't harm at least

Comment (Jojo-Schmitz) [reply CMakeLists.txt]: What abot the `/W4`?

File: CMakeLists.txt (hunk 1 above)
Comment (dmitrio95) [reply CMakeLists.txt]: Added this flag here

File: CMakeLists.txt (hunk 2 above)
Comment (dmitrio95) [reply CMakeLists.txt]: Added it

**Pull request merged**